### DISCUSSION TITLE
lombok plugin annotations

### DISCUSSION THREAD

**Discussion by Rasta341** [nvim-java/nvim-java#174] [Q&A] [answered]
uhm, hi :) can someone explain how use lombok annotations like "getter"/"setter" in code? 
i check on working project, but in  nvim its dont work with just annotations
I had to declare them explicitly for the code to work
![2024-05-04_14-29](https://github.com/nvim-java/nvim-java/assets/102875180/4f56a251-e9fa-482a-9463-64064041710a)
oh, yeah, and one more question about supporting some kind of logger like a @Slf4j

**Comment by s1n7ax** [comment]
@Rasta341 What do you mean it doesn't work? What happens when you don't have getters setters defined manually?

**Reply by Rasta341** [reply]
well, i was try launch simple spring app from this repo: https://github.com/DmitrijsFinaskins/simple-telegram-bot, in intelij idea all works fine, but when i try run in nvim, method getAd() "is not defined", and "log" - a variable of @slf4j highlighted as error... sorry, but right now cant give more information, if u have time, u can try launch code from that repo and looks by yourself :)

aaand thanks very much for answer)

**Reply by s1n7ax** [reply, marked as answer]
@Rasta341 It works perfectly fine. gd on getter or setter even takes me to the property defined. It's likely that nvim-java is not configured correctly.

**Reply by Rasta341** [reply]
Oh... well, sorry to bother you, looks like im doing something wrong :) thx for help

**Reply by beholders-eye** [reply]
> @Rasta341 It works perfectly fine. gd on getter or setter even takes me to the property defined. It's likely that nvim-java is not configured correctly.

No it doesn't work... sorry to say that.

**Reply by logrusx** [reply]
> > @Rasta341 It works perfectly fine. gd on getter or setter even takes me to the property defined. It's likely that nvim-java is not configured correctly.
> 
> No it doesn't work... sorry to say that.

At the moment when it was answered, it may very well have worked. I can confirm it worked a few months ago. However some things have changed since then.

Most notably, Mason has a 2.0 release which changed API.

Either way, it's counter productive to join in an already answered discussions. 

Most likely the problem will be solved with the migration to Mason 2.0 is completed. I'm currently stuck on it, but I and I guess others are working on it. 

Until then you can use the workaround mentioned in the Mason 2.0 migration issue.

**Reply by logrusx** [reply]
@beholders-eye [here](https://github.com/nvim-java/nvim-java/issues/384#issuecomment-3004645687)
